### PR TITLE
Fix: `StorageAccessException` shown after reinstall (API 30, non-Play Store build)

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
Scenario:
* The user can request 'MANAGE_EXTERNAL_STORAGE' (non-Play build)
* The collection path is `/storage/emulated/0/AnkiDroid`
* The user is on API 30
* The user reinstalled & kept data

Expected Result:
* We require 'MANAGE_EXTERNAL_STORAGE' to continue to access data

Previously:
1. We requested READ/WRITE_EXTERNAL_STORAGE
2. When accepted: 'StorageAccessException': User no longer had access to legacy storage.

## Approach
This splits `hasLegacyStoragePermissions` into two concepts:

1. Do we have a legacy collection
2. Does Android allow us to use legacy folders

Previously we only checked for (1) whereas we needed both

## How Has This Been Tested?
Unit tests + on API 30 emulators (`[amazon/Play]Debug` split)

## Learning

This builds on PR #13630: (ce0b0dc9ff6cfad16060e90e5e2f7316a8d478b0). Designed to fix https://github.com/ankidroid/Anki-Android/issues/13598

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison** [comment]
This builds on PR #13630: (ce0b0dc9ff6cfad16060e90e5e2f7316a8d478b0). Designed to fix https://github.com/ankidroid/Anki-Android/issues/13598

File: AnkiDroid/src/main/java/com/ichi2/anki/InitialActivity.kt [hunk 1]
```diff
@@ -281,9 +285,12 @@ class StartupStoragePermissionManager private constructor(
         }
 
         fun selectAnkiDroidFolder(context: Context): AnkiDroidFolder {
+            val canAccessLegacyStorage = Build.VERSION.SDK_INT < Build.VERSION_CODES.Q || Environment.isExternalStorageLegacy()
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/InitialActivity.kt:288]: Note: https://github.com/ankidroid/Anki-Android/issues/13654. This is incorrect on rare occasions when setting flag in debug builds

**Review by mikehardy — APPROVED** [review]
Looks good, marking for second review but second reviewer: no nitpicks or naming proposals or "I'd like it written this way please" - if the logic is correct and it does what it needs to approve and merge please.

**Review by BrayanDSO — APPROVED** [review]
Good catch. The PR LGTM. The tests are a nice touch

**Pull request merged**